src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -190,17 +218,21 @@ private Predicate addDateFilter(FiltersActivityDTO filtersActivityDTO, Predicate
     }
 
     @SuppressWarnings("checkstyle:cyclomaticComplexity")
-    private <T> void addFilters(CriteriaQuery<T> query, Root<ActivityInfo> root, UUID ownerId,
-        String searchName, SearchOn searchOn, FiltersActivityDTO filtersActivityDTO) {
-
-        Predicate predicate = cb.equal(root.get(ACCOUNT_ID), ownerId);
+    private <T> Predicate addFilters(CriteriaQuery<T> query, Root<ActivityInfo> root, UUID ownerId,
+        String searchName, SearchOn searchOn, FiltersActivityDTO filtersActivityDTO, Boolean partnerRecords) {
+
+        Predicate predicate;
+        if (partnerRecords) {
+            predicate = cb.notEqual(root.get(ACCOUNT_ID), ownerId);
```

Comment:
why is that? when looking for partner records, we don't want main organization to be from the current account

Reply:
but you're not looking only for partner records, right?
in case the partner records is true you're calling this method twice which will duplicate the name and location search, you can simplify this